tests/test_recombination_map.py

```diff
@@ -396,34 +440,76 @@ def test_mean_recombination_rate(self):
         mean_rr = recomb_map.mean_recombination_rate
         assert mean_rr == 0.0
 
-        # Test mean_recombination_rate is correct after reading from
-        # a hapmap file. read_hapmap() ignores the cM
-        # field, so here we test against using the cM field directly.
-        def hapmap_rr(hapmap_file):
-            first_pos = 0
+        # Test mean_recombination_rate is correct after reading from a hapmap file.
+        # read_hapmap() ignores the rate field, so here we test against the weighted
+        # mean from the rate field (converted to M/bp) directly. This only works if
+        # the first pos_cm is 0.0 (otherwise we will be missing the first rec rate)
+        def hapmap_mean(hapmap_file):
             with open(hapmap_file) as f:
+                prev_pos = 0
+                weights = []
+                rates = []
                 next(f)  # skip header
                 for line in f:
                     pos, rate, cM = map(float, line.split()[1:4])
                     if cM == 0:
-                        first_pos = pos
-            return cM / 100 / (pos - first_pos)
-
-        hapmap = """chr pos        rate                    cM
-                    1   4283592    3.79115663174456        0
-                    1   4361401    0.0664276817058413      0.294986106359414
-                    1   7979763   10.9082897515584         0.535345505591925
-                    1   8007051    0.0976780648822495      0.833010916332456
-                    1   8762788    0.0899929572085616      0.906829844052373
-                    1   9477943    0.0864382908650907      0.971188757364862
-                    1   9696341    4.76495005895746        0.990066707213216
-                    1   9752154    0.0864316558730679      1.25601286485381
-                    1   9881751    0.0                     1.26721414815999"""
+                        prev_rate = 0
+                    weights.append(pos - prev_pos)
+                    rates.append(prev_rate)
+                    prev_pos = pos
+                    prev_rate = rate
+            return np.average(rates, weights=weights) * 1e-8  # convert to morgans / bp
```

Comment:
This weighted average produces incorrect values, because it fails to account for zero-rate regions at the start of files. I believe the existing code in this test was producing correct values.

Reply:
Hmm. The start of the files may not be zero-rate. It depends if the first genetic position is 0cM. I'll look into the regression and see what the issue is.